src/wallet/scriptpubkeyman.cpp

```diff
@@ -1616,6 +1616,13 @@ bool DescriptorScriptPubKeyMan::GetReservedDestination(const OutputType type, bo
 
 void DescriptorScriptPubKeyMan::ReturnDestination(int64_t index, bool internal, const CTxDestination& addr)
 {
+    LOCK(cs_desc_man);
+    // Only return when the index was the most recent
```

Comment:
nit: maybe swap eb9c41e9240fe01f6404cf23abe1fc35ca146586 and ee03d46178a3e8d511c33f734892820b36ed7a1a.

A test would be nice.

Reply:
Swapped the commits.

A general test for `GetReserveDestination` and `ReturnDestination` in the wallet feels out of scope for this PR. IMO should be a separate PR.

Reply:
Doesn't `walletcreatefundedpsbt` call those two methods? So the test just has to check the keypool doesn't grow.

Reply:
Again, seems out of scope and a general wallet test rather than specific to descriptor wallets.